src/classes/ALLO_Allocations_TDTM.cls

```diff
@@ -538,8 +538,9 @@ public class ALLO_Allocations_TDTM extends TDTM_Runnable {
         for (SObject sObj : listSObjs) {
 
             //if we have recurring donation or campaign allocations
-            if ((sObj instanceof npe01__OppPayment__c && mapWrapper.containsKey((Id)sObj.get('npe01__Opportunity__c'))) || !(sObj instanceof npe01__OppPayment__c) && ((sObj.get('npe03__Recurring_Donation__c') != null && mapWrapper.containsKey((Id)sObj.get('npe03__Recurring_Donation__c'))) ||
-                (sObj.get('CampaignId') != null && mapWrapper.containsKey((Id)sObj.get('CampaignId'))))) {
+            if ((sObj instanceof npe01__OppPayment__c && mapWrapper.containsKey((Id)sObj.get('npe01__Opportunity__c'))) ||
```

Comment:
Wow, this method is really hard to read and decipher. In addition to the bug fix, it might be helpful to slightly refactor:

```
    for (SObject sObj : listSObjs) {
        Boolean isPaymentRecord = sObj instanceof npe01__OppPayment__c;
        Boolean isOpportunityRecord = sObj instanceof Opportunity;

        if (isPaymentRecord && mapWrapper.containsKey((Id)sObj.get('npe01__Opportunity__c'))) {
            copyRecurringDonationCampaignAndPaymentAllocations(sObj);

        } else if (isOpportunityRecord && sObj.get('npe03__Recurring_Donation__c') != null && mapWrapper.containsKey((Id)sObj.get('npe03__Recurring_Donation__c')) {
            copyRecurringDonationCampaignAndPaymentAllocations(sObj);

        } else if (!isOpportunityRecord && sObj.get('CampaignId') != null && mapWrapper.containsKey((Id)sObj.get('CampaignId')) {
            copyRecurringDonationCampaignAndPaymentAllocations(sObj);

        } else if (settings.Default_Allocations_Enabled__c) {
            dmlWrapper.objectsToInsert.add(makeDefaultAllocation(sObj)); 
        }
    }

.....

private void  copyRecurringDonationCampaignAndPaymentAllocations(SObject) {
    // copy everything in that if block from line 545 - 680
}
```

Reply:
I get where you are headed and will do that, but I believe it needs a little tweaking...

Reply:
no tweaking needed .  bad first reading on my part.  Nice!